### PR TITLE
[bugfix](paimon)add support for 'in' and 'not in'

### PR DESCRIPTION
**Pull request by wuwenchi**
## Proposed changes

add support for `in` and `not in`:

```
select * from tb where partition_column in ('a','b','c');
select * from tb where partition_column not in ('a','b','c');

```

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/source/PaimonScanNode.java [hunk 1]
```diff
@@ -317,15 +317,26 @@ public Map<String, String> getLocationProperties() throws MetaNotFoundException,
...
     @Override
     public String getNodeExplainString(String prefix, TExplainLevel detailLevel) {
-        String result = super.getNodeExplainString(prefix, detailLevel)
-                + String.format("%spaimonNativeReadSplits=%d/%d\n",
-                        prefix, rawFileSplitNum, (paimonSplitNum + rawFileSplitNum));
+        StringBuilder sb = new StringBuilder(super.getNodeExplainString(prefix, detailLevel));
+        sb.append(String.format("%spaimonNativeReadSplits=%d/%d\n",
+                prefix, rawFileSplitNum, (paimonSplitNum + rawFileSplitNum)));
+
+        sb.append(prefix).append("spredicateFromPaimon:");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/source/PaimonScanNode.java:324]:
```suggestion
        sb.append(prefix).append("predicatesFromPaimon:");
```

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by hubgeter — APPROVED** [review]
LGTM

**Pull request merged**